File: domox-service/src/main/kotlin/domox/nlp/SentenceTO.java
package domox.nlp;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;

import java.util.List;

@Data
public class SentenceTO {
    Long index = 0L;
    String parse = "";
    List<BasicDependencyTO> basicDependencies = null;
    List<BasicDependencyTO> enhancedDependencies = null;
    List<BasicDependencyTO> enhancedPlusPlusDependencies = null;
    String sentimentValue = "";
    String sentiment = "";
    List<Long> sentimentDistribution = null;
    String sentimentTree = "";
    @JsonIgnore
    List<EntityMentionTO> entitymentions = null;
    public List<TokenTO> tokens = null;
}

File: domox-service/src/main/kotlin/domox/nlp/SentenceTO.kt
package domox.nlp

import com.fasterxml.jackson.annotation.JsonIgnore

data class SentenceTO(
    val index: Long = 0L,
    val parse: String = "",
    val basicDependencies: List<BasicDependencyTO> = emptyList(),
    val enhancedDependencies: List<BasicDependencyTO> = emptyList(),
    val enhancedPlusPlusDependencies: List<BasicDependencyTO> = emptyList(),
    val sentimentValue: String = "",
    val sentiment: String = "",
    val sentimentDistribution: List<Long> = emptyList(),
    val sentimentTree: String = "",
    @JsonIgnore
    val entitymentions: List<EntityMentionTO>? = null,
    val tokens: List<TokenTO> = emptyList()
)
